src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SourceTextExtensions.cs

```diff
@@ -145,5 +145,33 @@ public static bool NonWhitespaceContentEquals(this SourceText source, SourceText
 
             return null;
         }
+
+        public static int? GetLastNonWhitespaceOffset(this SourceText source, TextSpan? span, out int newLineCount)
```

Comment:
I think this needs a bit of a doc-comment, I'm looking at the surrounding PR and having a little trouble telling what the goal is.